ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1103 @@
+ * */
+
+'use strict';
+
+import type {
+    AlignValue,
+    VerticalAlignValue
+} from '../Core/Renderer/AlignObject';
+import type {
+    CSSObject
+} from '../Core/Renderer/CSSObject';
+import type SVGElement from '../Core/Renderer/SVG/SVGElement';
+import Chart from '../Core/Chart/Chart.js';
+import H from '../Core/Globals.js';
+import D from '../Core/DefaultOptions.js';
+const { defaultOptions } = D;
+import U from '../Core/Utilities.js';
+import Point from '../Core/Series/Point';
+import type TreemapSeries from '../Series/Treemap/TreemapSeries';
+import SeriesType from '../Core/Series/SeriesType';
+import F from '../Core/FormatUtilities.js';
+import SVGAttributes from '../Core/Renderer/SVG/SVGAttributes';
+const {
+    format
+} = F;
+const {
+    addEvent,
+    defined,
+    extend,
+    merge
+} = U;
+
+/* *
+ *
+ * Declarations
+ *
+ * */
+
+declare module '../Core/LangOptions' {
+    interface LangOptions {
+        mainBreadcrumb?: string;
+    }
+}
+declare module '../Core/Chart/ChartLike' {
+    interface ChartLike {
+        breadcrumbs?: Breadcrumbs;
+    }
+}
+declare module '../Series/Treemap/TreemapSeriesOptions' {
+    interface TreemapSeriesOptions {
+        breadcrumbs?: Highcharts.BreadcrumbsOptions;
+    }
+}
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        interface BreadcrumbsOptions {
+            buttonPadding?: number;
+            events?: BreadcrumbsButtonsEventsOptions;
+            floating?: boolean;
+            format?: string;
+            formatter?: BreadcrumbsButtonsFormatter;
+            position?: BreadcrumbsAlignOptions;
+            separator?: SeparatorOptions;
+            showFullPath?: boolean;
+            style?: CSSObject;
+            useHTML?: boolean;
+            zIndex?: number;
+        }
+        interface BreadcrumbsAlignOptions {
+            align?: AlignValue;
+            verticalAlign?: VerticalAlignValue;
+            alignByTranslate?: boolean;
+            x: number;
+            y: number;
+        }
+        interface BreadcrumbsClickCallbackFunction {
+            (e: Event, breadcrumbs: Breadcrumbs): (boolean|undefined);
+        }
+        interface BreadcrumbsButtonsEventsOptions {
+            click?: BreadcrumbsClickCallbackFunction;
+        }
+        interface BreadcrumbsButtonsFormatter {
+            (
+                breadcrumb: Array<number|null|Point|SeriesType|SVGAElement>,
+                breadcrumbs: Breadcrumbs
+            ): (string);
+        }
+        interface DrilldownOptions {
+            breadcrumbs?: BreadcrumbsOptions;
+        }
+        interface SeparatorOptions {
+            symbol?: string;
+            size?: number;
+        }
+        interface TreemapSeriesOptions {
+            breadcrumbs?: BreadcrumbsOptions;
+        }
+        interface ChartLike {
+            breadcrumbs?: Breadcrumbs;
+        }
+        interface LangOptions {
+            mainBreadcrumb?: string;
+        }
```

Comment:
The interfaces should be part of `declare module` patterns, or in case of breadcrumb-specific go into the class `namespace Breadcrumbs` right before the default export.

In general nothing new should be added to the internal namespace. It also has a deprecated tag for some time.